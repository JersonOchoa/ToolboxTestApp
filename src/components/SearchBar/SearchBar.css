.searchBar {
    background-color: red;
    display: flex;
    flex-direction: row;
    justify-content: center;
    height: 10vh;
}

.searchBar__form {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 10px;
    width: 50%;
}

.searchBar__form Button {
    margin-left: 25px;
}
