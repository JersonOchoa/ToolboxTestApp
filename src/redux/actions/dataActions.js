import axiosInstance from "../../util/axios";
import { requestsTest } from "../../util/requests";
import { ADD_RESULT } from "../actions";

export const reverseText = (originalText) => (dispatch) => {
    axiosInstance
        .post(requestsTest.fetchTextReversed, { text: originalText })
        .then((res) => {
            const { text, palindrome } = res.data;
            dispatch({
                type: ADD_RESULT,
                payload: [
                    {
                        text: originalText,
                        reversed: text,
                        palindrome: palindrome.toString(),
                    },
                ],
            });
        })
        .catch((err) => console.log(err));
};
