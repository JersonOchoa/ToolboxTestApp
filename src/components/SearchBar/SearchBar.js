import React, { useState } from "react";
import { Button, Form } from "react-bootstrap";
import { connect } from "react-redux";
import { reverseText } from "../../redux/actions/dataActions";
//CSS
import "bootstrap/dist/css/bootstrap.min.css";
import "./SearchBar.css";

const SearchBar = (props) => {
    const { reverseText } = props;

    const [text, setText] = useState("");

    const handleSubmit = (e) => {
        e.preventDefault();
        reverseText(text);
        setText("");
    };

    return (
        <div className="searchBar">
            <Form className="searchBar__form" onSubmit={handleSubmit}>
                <Form.Control
                    type="text"
                    placeholder="Insert Text"
                    value={text}
                    onChange={(e) => setText(e.target.value)}
                />
                <Button variant="primary" type="submit">
                    Send
                </Button>
            </Form>
        </div>
    );
};

const mapActionsToProps = {
    reverseText,
};

export default connect(null, mapActionsToProps)(SearchBar);
