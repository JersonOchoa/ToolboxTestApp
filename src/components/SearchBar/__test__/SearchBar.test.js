import React from "react";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import SearchBar from "../SearchBar";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { reverseText } from "../../../redux/actions/dataActions";

const mockStore = configureStore();

describe("React Redux component", () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({ listResults: [] });

        store.dispatch = jest.fn();

        component = renderer.create(
            <Provider store={store}>
                <SearchBar />
            </Provider>
        );
    });

    it("Render without crashing", () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
});
