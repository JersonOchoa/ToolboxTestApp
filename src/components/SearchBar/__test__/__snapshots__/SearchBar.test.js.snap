// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Redux component Render without crashing 1`] = `
<div
  className="searchBar"
>
  <form
    className="searchBar__form"
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Insert Text"
      type="text"
      value=""
    />
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Send
    </button>
  </form>
</div>
`;
